kernel/rtlil.cc

```diff
@@ -4031,16 +4031,20 @@ void RTLIL::SigSpec::replace(const RTLIL::SigSpec &pattern, const RTLIL::SigSpec
 	unpack();
 	other->unpack();
 
+	dict<RTLIL::SigBit, int> pattern_to_with;
```

Comment:
I guess if a dict is what it takes, then a dict is what it takes.

Reply:
It surprised a few of my coworkers as well. That's why real measurement is so important to optimization